modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-channel-impl/src/main/java/com/liferay/headless/commerce/admin/channel/internal/resource/v1_0/CategoryDisplayPageResourceImpl.java

```diff
@@ -0,0 +1,229 @@
+					CPDisplayLayoutConstants.TYPE_LAYOUT, search,
+					pagination.getStartPosition(), pagination.getEndPosition(),
+					sort);
+
+		return Page.of(
+			HashMapBuilder.put(
+				"post",
+				addAction(
+					ActionKeys.UPDATE, id, "postChannelIdCategoryDisplayPage",
+					commerceChannel.getUserId(),
+					CommerceChannel.class.getName(),
+					commerceChannel.getGroupId())
+			).build(),
+			transform(
+				cpDisplayLayoutBaseModelSearchResult.getBaseModels(),
+				cpDisplayLayout -> _toCategoryDisplayPage(
+					cpDisplayLayout.getCPDisplayLayoutId())));
+	}
+
+	public CategoryDisplayPage patchCategoryDisplayPage(
+			Long id, CategoryDisplayPage categoryDisplayPage)
+		throws Exception {
+
+		CPDisplayLayout cpDisplayLayout =
+			_cpDisplayLayoutService.fetchCPDisplayLayout(id);
+
+		if (cpDisplayLayout == null) {
+			throw new NoSuchCPDisplayLayoutException();
+		}
+
+		_cpDisplayLayoutService.updateCPDisplayLayout(
+			id,
+			GetterUtil.getLong(
+				categoryDisplayPage.getCategoryId(),
+				cpDisplayLayout.getClassPK()),
+			cpDisplayLayout.getLayoutPageTemplateEntryUuid(),
+			GetterUtil.getString(
+				categoryDisplayPage.getPageUuid(),
+				cpDisplayLayout.getLayoutUuid()));
+
+		return _toCategoryDisplayPage(id);
+	}
+
+	public CategoryDisplayPage
+			postChannelByExternalReferenceCodeCategoryDisplayPage(
+				String externalReferenceCode,
+				CategoryDisplayPage categoryDisplayPage)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.
+				getCommerceChannelByExternalReferenceCode(
+					externalReferenceCode, contextCompany.getCompanyId());
+
+		return postChannelIdCategoryDisplayPage(
+			commerceChannel.getCommerceChannelId(), categoryDisplayPage);
+	}
+
+	public CategoryDisplayPage postChannelIdCategoryDisplayPage(
+			Long id, CategoryDisplayPage categoryDisplayPage)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannel(id);
+
+		CPDisplayLayout cpDisplayLayout =
+			_cpDisplayLayoutService.addCPDisplayLayout(
+				commerceChannel.getSiteGroupId(), AssetCategory.class,
+				categoryDisplayPage.getCategoryId(), null,
+				categoryDisplayPage.getPageUuid());
+
+		return _toCategoryDisplayPage(cpDisplayLayout.getCPDisplayLayoutId());
+	}
+
+	private Map<String, Map<String, String>> _getActions(
+		CPDisplayLayout cpDisplayLayout) {
+
+		return HashMapBuilder.<String, Map<String, String>>put(
+			"delete",
+			addAction(
+				ActionKeys.DELETE, cpDisplayLayout.getCPDisplayLayoutId(),
+				"deleteCategoryDisplayPage",
+				_cpDisplayLayoutModelResourcePermission)
+		).put(
+			"get",
+			addAction(
+				ActionKeys.VIEW, cpDisplayLayout.getCPDisplayLayoutId(),
+				"getCategoryDisplayPage",
+				_cpDisplayLayoutModelResourcePermission)
+		).put(
+			"patch",
+			addAction(
+				ActionKeys.UPDATE, cpDisplayLayout.getCPDisplayLayoutId(),
+				"patchCategoryDisplayPage",
+				_cpDisplayLayoutModelResourcePermission)
+		).build();
+	}
```

Comment:
Let's use directly the ModelResourcePermission of the assetCategory instead of `_cpDisplayLayoutModelResourcePermission`